scripts/extensions/sams/src/components/assets/assetGridItem.tsx

```diff
@@ -94,7 +95,8 @@ export class AssetGridItem extends React.PureComponent<IProps> {
             <GridItem
                 onClick={this.onItemClick}
                 onDoubleClick={this.onItemDoubleClick}
-                selected={this.props.selected || this.props.itemSelected}
+                singleClickSelected={this.props.selected || this.props.itemSelected}
+                locked={this.props.selected && this.props.itemSelectedLocked}
```

Comment:
The red border should always show if the Asset is locked, regardless if the item is selected or not (this goes for both Grid and List item views)

Reply:
Also we could identify the lock state inside `assetGridItem` and `assetListItem` components directly by checking `lock_session` has a value. So we wouldn't need to send through `itemSelectedLocked`. As this visual lock indicates that the actual Asset in the SAMS API is locked